scripts/marketing_data_test.py

```diff
@@ -0,0 +1,481 @@
+
+    dtstr = datetime.now().strftime("%Y%m%dT%H%M%S")
+    result_file = data_file.replace('.joblib', f'_{model_name}_results_{dtstr}.csv')
+    if not os.path.isfile(result_file):
+        with open(result_file, 'w') as output_results:
+            for k in dml_cv_param_dict.keys():
+                output_results.write(f"DML_{k},")
+            for k in y_cv_param_dict.keys():
+                output_results.write(f"Y_{k},")
+            for k in t_cv_param_dict.keys():
+                output_results.write(f"T_{k},")
+            for f_idx in range(0, cv_folds):
+                output_results.write(f"F{f_idx}_y_score,")
+            for f_idx in range(0, cv_folds):
+                for tidx in range(T.shape[1]):
+                    output_results.write(f"F{f_idx}_T{tidx}_score,")
+            output_results.write("Yresid_RMSE,Yresid_r2,Yresid_corc,Yresid_corp\n")
+
+    output_results = open(result_file, 'a')
+
+    # treat_est_df = treat_df.drop('treat_id', axis=1)
+    # Effect of the treatments trained on
+    x_cols = list(X.columns.values)
+    t_cols = list(T.columns.values)
+    X['x_id']=X.index.values
+
+    n_sample = 10000
+    if X.shape[0] > n_sample:
+        treat_est_combo = X.sample(n=n_sample).reset_index().merge(treat_df, how='cross')
+    else:
+        treat_est_combo = X.merge(treat_df, how='cross')
+
+    logger.info(f"""
+        Calculating effects of {len(treat_df)} treatments on down-sampled audience size of
+        {n_sample}, with total {len(treat_est_combo)} combinations.
+        """
+    )
+    # Save the combined sample identifiers, and then remove them
+    treat_identifiers = treat_est_combo[['x_id','treat_id']]
+    treat_est_combo = treat_est_combo.drop(['x_id','treat_id'],axis=1)
+    X = X.drop(['x_id'],axis=1)
+
+    # Simple un-stratified train/test split - need to redo this ASAP
+
+    if do_evaluate:
+        ind = np.array(range(len(X)))
+        train_ind = np.random.choice(len(X), int((1.0-val_pcnt)*len(X)), replace=False)
+        val_ind = ind[~np.isin(ind, train_ind)]
+        Xtrain = X.iloc[train_ind,:]
+        Ttrain = T.iloc[train_ind,:]
+        ytrain = y[train_ind]
+        Xval = X.iloc[val_ind,:]
+        Tval = T.iloc[val_ind,:]
+        yval = y[val_ind]
+    else:
+        Xtrain = X
+        Ttrain = T
+        ytrain = y
+
+
+    for y_param_combo in product(*y_cv_param_dict.values()):
+        y_params = dict(zip(y_cv_param_dict.keys(), y_param_combo))
+
+        logger.info("#############################################################################")
+
+        logger.info(f"Fitting the standard (confounded) model with params {y_params}")
+        noncausal_est = XGBClassAUCScore(**y_params)
+        # TODO: Confirm with Carl that redefining treat_est_combo is fine since we removed
+        # collinear features.
+        treat_est_combo = pd.concat([X,T], axis=1)
+        noncausal_est.fit(treat_est_combo, y)
```

Comment:
Do we need to fit standard model? We will be fitting `CausalForestDML()` below, and based on the [documentation](https://econml.azurewebsites.net/_autosummary/econml.dml.CausalForestDML.html), it has parameter `honest = True` by default. This means the `est = CausalForestDML(...)` will train by splitting into 2 equal subsets anyway. I'm basing the definition of honesty from this paper [here](https://arxiv.org/pdf/1510.04342) (section 2.4).